### PR TITLE
Fix regression tests paths

### PR DESCRIPTION
**Pull request by cmouse**
Allows regression testing for out-of-tree builds

### CODE REVIEW

**Comment by rubenk** [comment]
can you squash 60a200a and fd73b6b together?

**Comment by cmouse** [comment]
I'll squash it if travis is happy.

**Comment by pieterlexis** [comment]
looks like the moduledir is wrong:

`[2015-04-30 11:49:12] Apr 30 11:49:12 Unable to load module '../modules/libbindbackend.so': ../modules/libbindbackend.so: cannot open shared object file: No such file or directory`

**Comment by cmouse** [comment]
Ready for review.

**Comment by cmouse** [comment]
I have no idea why travis has failed here, it only indicates that it stalled at the end.

**Comment by mind04** [comment]
i give you one hint, removed & and \ at the end of various lines

**Comment by cmouse** [comment]
Power of sed. Thanks.

**Comment by cmouse** [comment]
My my. Travis finally agrees. Ready for review again.

**Comment by rubenk** [comment]
Looks good to me.

**Comment by mind04** [comment]
- cleanup travis.yml, module symlinks are created in start-test-stop and if you need this here something else is missing
- exclude .gitkeep when the modules dir is cleared
- add line in .gitignore for module symlinks
- switch to out of tree testing in travis (move test directories one level op and run the test from the new location)

**Comment by cmouse** [comment]
I am excluding .gitkeep =) not sure what you mean...

I was thinking someone smarter would do the out of tree build change for travis, I am not sure how to do it.

**Comment by pieterlexis** [comment]
Out of tree building would be nice indeed, that way we know if we have all the dependencies for the tarballs right, but imho could a different PR. Judging by the changes in `.travis.yml`, .gitkeep will stay where it is :). So.. LGTM :+1:

**Comment by pieterlexis** [comment]
@Habbie any comments?

File: regression-tests/test.vars [hunk 1]
```diff
@@ -0,0 +1,14 @@
+BUILDDIR=`readlink -m ${BUILDDIR:-..}`
```
Comment (Habbie) [inline regression-tests/test.vars]: -m appears to be a GNU extension

**Comment by Habbie** [comment]
Other than the non-portable readlink invocation, I think this looks fine.

**Comment by cmouse** [comment]
Any suggestions on portable alternative? I'm going to ofc

**Comment by rubenk** [comment]
readlink -f doesn't work?

**Comment by cmouse** [comment]
It does not produce fully canonical paths. I tried. For some reason I ended up with ".." in the path. Can't remember the details now.

**Comment by rubenk** [comment]
perhaps `python -c 'import sys; import os.path; print os.path.realpath(sys.argv[1])'`?

**Comment by cmouse** [comment]
if that's portable I'm game, since we use python already. But I think that would go into some helper py script.

**Comment by rubenk** [comment]
I tested it on Linux and the BSDs so we should be fine. But yes, preferably in a helper script :)

**Comment by mind04** [comment]
Ok, I missed the different handling of files and .files in bash, but the other 3 points are still valid.
 
1 If powerdns is build with dyn-modules. The changes in travis.yml are hiding the "not being able to run (nobackend and api) tests without running start-test-stop first" problem. Why not leave the default module symlinks in place and only update them for out of tree testing?

2 my bad

3 there are untracked files after you run the regression tests
```
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#	modules/libbindbackend.so
#	modules/libgmysqlbackend.so
#	modules/libgpgsqlbackend.so
#	modules/libgsqlite3backend.so
#	modules/libldapbackend.so
#	modules/liblmdbbackend.so
#	modules/libluabackend.so
#	modules/libmydnsbackend.so
#	modules/librandombackend.so
```

4 I'm not talking about out of tree builds, but about out of tree testing which is fairly easy.
```
mv regression-tests* ../
cd ..
export BUILDDIR=$PWD
cd regerssion-tests.nobackend
./runtests
cd ../regression-tests
# run your tests
./start-test-stop ....
```
And if you do this, you should notice start-test-stop in regeression-tests and runtests in regerssion-tests.nobackend are broken (path problems).

**Comment by cmouse** [comment]
Um. I was more thinking that we'd run out of tree builds and run regression test suite against them. the test suite itself isn't relocatable, only the build is. You are not supposed to relocate the test suite.

**Comment by cmouse** [comment]
Also you are supposed to be able to run the other test suites against build in /tmp/build

**Comment by Habbie** [comment]
Concerns 1 and 3 appear valid to me (but just fixing 1 should cover that). 4 is valid, but only a blocker if this pull actually breaks that feature.

**Comment by cmouse** [comment]
There is now tmpmodules directory which is used solely by regression-tests, others use modules (for now).

**Comment by cmouse** [comment]
All the pointed out issues should be dealt with now. You can do out of tree testing and out of tree builds.

**Comment by cmouse** [comment]
@mind04 can you comment if you agree on this pr now?

**Comment by pieterlexis** [comment]
@mind04 any comments on this?

**Comment by cmouse** [comment]
Perhaps someone else can do this.

**Pull request closed without merging**